train.py

```diff
@@ -278,7 +281,7 @@ def run_train(
                 loss = loss / cfg.training.grad_accumulation
 
             # Backward pass
-            if cfg.environment.mixed_precision:
+            if cfg.environment.mixed_precision and not cfg.environment.use_deepspeed:
```

Comment:
How exactly does deepspeed handle mixed precision training?

Reply:
When use deepspeed and type fp16 or bf16, it will do mixed precision automatically.

https://huggingface.co/docs/transformers/main_classes/deepspeed#automatic-mixed-precision

Maybe we can warning it and force mixed precision to be True when using deepspeed?